### ISSUE TITLE
How to set the weight decay decrease by per number of epoch?

### ISSUE THREAD

**Issue by Keyslime** [open-mmlab/mmdetection#6210]
I want to have an optimizer of SGD with an initial learning rate of 0.001, and the learning rate is dropped by a factor of 10 every 10 epochs in my RetinaNet Model, but I am not quite sure where can it be configured in the config where every 10 epoch it drops by a factor of 10.  Currently, I have this. 
`optimizer= dict(type='SGD', lr=0.001, momentum=0.9, weight_decay=0.0001)`

Do I change the weight_decay number to 10? Sorry still kinda new with machine learning.

**Comment by AndreaPi** [comment]
Hi,

disclaimer: not one of the developers (just a very happy user 😊 ). What you're describing is not weight decay (which is a form or regularization which adds a penalty term to the loss in order to encourage the model to select smaller or sparser weights, depending on whether you use L1 or L2 weight decay). You're talking about _learning rate decay_ (also called _learning rate annealing_, or _learning rate schedule_). There are many different learning rate schedules: the one you requested (divide the learning rate value by 10 every 10 epochs) can be very easily implemented in MMDetection adding this code to your config file:

```
lr_config = dict(  
    policy='step',  
    warmup='linear',  
    warmup_iters=500, 
    warmup_ratio=0.001,  
    step=10) 
```

I _think_ you can also change the reduction factor from 0.1 to, say, 0.05, this way:

```
lr_config = dict(  
    policy='step',  
    warmup='linear',  
    warmup_iters=500, 
    warmup_ratio=0.001,  
    gamma=0.05,
    step=10) 
```

To get more info on this stuff, you can read here:

https://mmdetection.readthedocs.io/en/latest/tutorials/config.html?highlight=lr_config#an-example-of-mask-r-cnn

https://mmdetection.readthedocs.io/en/latest/tutorials/customize_runtime.html?highlight=lr_config

https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/lr_updater.py

If I were in you, however, I'd study the foundations of ML more, before starting to use a Deep Learning-based object detection framework.

**Comment by Keyslime** [comment]
Sorry, I misspoke. But thanks for the tip! I will try to read more about it.

**Comment by AndreaPi** [comment]
BTW, if the answer worked for you, I think you can close the issue.
